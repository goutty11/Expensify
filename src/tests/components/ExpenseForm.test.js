import React from 'react';
import { shallow } from 'enzyme';
import ExpenseForm from '../../components/ExpenseForm';
import expense from '../fixtures/expenses';
import toJson from 'enzyme-to-json';

test('should render ExpenseForm correctly', () => {
    const wrapper = shallow(<ExpenseForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

// test('should render ExpenseForm correctly with expense date', () => {
//     const wrapper = shallow(<ExpenseForm expense={expenses[1]}/>);
//     expect(wrapper).toMatchSnapshot();
// });

