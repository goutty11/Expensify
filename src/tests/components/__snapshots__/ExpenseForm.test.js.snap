// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseForm correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "",
        <form
          onSubmit={[Function]}
>
          <input
                    autoFocus={true}
                    onChange={[Function]}
                    placeholder="Description"
                    type="text"
                    value=""
          />
          <input
                    onChange={[Function]}
                    placeholder="Amount"
                    type="text"
                    value=""
          />
          <DatePicker
                    allowSameDay={false}
                    dateFormat="L"
                    dateFormatCalendar="MMMM YYYY"
                    disabled={false}
                    disabledKeyboardNavigation={false}
                    dropdownMode="scroll"
                    monthsShown={1}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClickOutside={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMonthChange={[Function]}
                    onSelect={[Function]}
                    onYearChange={[Function]}
                    preventOpenOnFocus={false}
                    selected={"2018-02-25T06:47:00.702Z"}
                    shouldCloseOnSelect={true}
                    showTimeSelect={false}
                    timeIntervals={30}
                    withPortal={false}
          />
          <textarea
                    onChange={[Function]}
                    placeholder="Add a note for your expense (optional)"
                    value=""
          />
          <button>
                    Add Expense
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              autoFocus={true}
              onChange={[Function]}
              placeholder="Description"
              type="text"
              value=""
/>,
            <input
              onChange={[Function]}
              placeholder="Amount"
              type="text"
              value=""
/>,
            <DatePicker
              allowSameDay={false}
              dateFormat="L"
              dateFormatCalendar="MMMM YYYY"
              disabled={false}
              disabledKeyboardNavigation={false}
              dropdownMode="scroll"
              monthsShown={1}
              onBlur={[Function]}
              onChange={[Function]}
              onClickOutside={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMonthChange={[Function]}
              onSelect={[Function]}
              onYearChange={[Function]}
              preventOpenOnFocus={false}
              selected={"2018-02-25T06:47:00.702Z"}
              shouldCloseOnSelect={true}
              showTimeSelect={false}
              timeIntervals={30}
              withPortal={false}
/>,
            <textarea
              onChange={[Function]}
              placeholder="Add a note for your expense (optional)"
              value=""
/>,
            <button>
              Add Expense
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": true,
              "onChange": [Function],
              "placeholder": "Description",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Amount",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowSameDay": false,
              "dateFormat": "L",
              "dateFormatCalendar": "MMMM YYYY",
              "disabled": false,
              "disabledKeyboardNavigation": false,
              "dropdownMode": "scroll",
              "monthsShown": 1,
              "onBlur": [Function],
              "onChange": [Function],
              "onClickOutside": [Function],
              "onFocus": [Function],
              "onKeyDown": [Function],
              "onMonthChange": [Function],
              "onSelect": [Function],
              "onYearChange": [Function],
              "preventOpenOnFocus": false,
              "selected": "2018-02-25T06:47:00.702Z",
              "shouldCloseOnSelect": true,
              "showTimeSelect": false,
              "timeIntervals": 30,
              "withPortal": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Add a note for your expense (optional)",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Expense",
            },
            "ref": null,
            "rendered": "Add Expense",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <form
            onSubmit={[Function]}
>
            <input
                        autoFocus={true}
                        onChange={[Function]}
                        placeholder="Description"
                        type="text"
                        value=""
            />
            <input
                        onChange={[Function]}
                        placeholder="Amount"
                        type="text"
                        value=""
            />
            <DatePicker
                        allowSameDay={false}
                        dateFormat="L"
                        dateFormatCalendar="MMMM YYYY"
                        disabled={false}
                        disabledKeyboardNavigation={false}
                        dropdownMode="scroll"
                        monthsShown={1}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClickOutside={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onMonthChange={[Function]}
                        onSelect={[Function]}
                        onYearChange={[Function]}
                        preventOpenOnFocus={false}
                        selected={"2018-02-25T06:47:00.702Z"}
                        shouldCloseOnSelect={true}
                        showTimeSelect={false}
                        timeIntervals={30}
                        withPortal={false}
            />
            <textarea
                        onChange={[Function]}
                        placeholder="Add a note for your expense (optional)"
                        value=""
            />
            <button>
                        Add Expense
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                autoFocus={true}
                onChange={[Function]}
                placeholder="Description"
                type="text"
                value=""
/>,
              <input
                onChange={[Function]}
                placeholder="Amount"
                type="text"
                value=""
/>,
              <DatePicker
                allowSameDay={false}
                dateFormat="L"
                dateFormatCalendar="MMMM YYYY"
                disabled={false}
                disabledKeyboardNavigation={false}
                dropdownMode="scroll"
                monthsShown={1}
                onBlur={[Function]}
                onChange={[Function]}
                onClickOutside={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMonthChange={[Function]}
                onSelect={[Function]}
                onYearChange={[Function]}
                preventOpenOnFocus={false}
                selected={"2018-02-25T06:47:00.702Z"}
                shouldCloseOnSelect={true}
                showTimeSelect={false}
                timeIntervals={30}
                withPortal={false}
/>,
              <textarea
                onChange={[Function]}
                placeholder="Add a note for your expense (optional)"
                value=""
/>,
              <button>
                Add Expense
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoFocus": true,
                "onChange": [Function],
                "placeholder": "Description",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Amount",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "allowSameDay": false,
                "dateFormat": "L",
                "dateFormatCalendar": "MMMM YYYY",
                "disabled": false,
                "disabledKeyboardNavigation": false,
                "dropdownMode": "scroll",
                "monthsShown": 1,
                "onBlur": [Function],
                "onChange": [Function],
                "onClickOutside": [Function],
                "onFocus": [Function],
                "onKeyDown": [Function],
                "onMonthChange": [Function],
                "onSelect": [Function],
                "onYearChange": [Function],
                "preventOpenOnFocus": false,
                "selected": "2018-02-25T06:47:00.702Z",
                "shouldCloseOnSelect": true,
                "showTimeSelect": false,
                "timeIntervals": 30,
                "withPortal": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Add a note for your expense (optional)",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Expense",
              },
              "ref": null,
              "rendered": "Add Expense",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
